crates/net/discv4/src/lib.rs

```diff
@@ -1079,25 +1111,13 @@ impl Discv4Service {
             // ping's expiration timestamp is in the past
             return
         }
+        if node_id == *self.local_peer_id() {
+            // ignore find node requests to ourselves
+            return
+        }
 
-        let key = kad_key(node_id);
-
-        match self.kbuckets.entry(&key) {
-            kbucket::Entry::Present(_, status) => {
-                if status.is_connected() {
-                    self.respond_closest(msg.id, remote_addr)
-                }
-            }
-            kbucket::Entry::Pending(_, status) => {
-                if status.is_connected() {
-                    self.respond_closest(msg.id, remote_addr)
-                }
-            }
-            kbucket::Entry::Absent(_) => {
-                // no existing endpoint proof
-                // > To guard against traffic amplification attacks, Neighbors replies should only be sent if the sender of FindNode has been verified by the endpoint proof procedure.
-            }
-            kbucket::Entry::SelfEntry => {}
+        if self.has_bond(node_id, remote_addr.ip()) {
+            self.respond_closest(msg.id, remote_addr)
         }
```

Comment:
No need to respond if `status.is_connected()` now? I guess it's implicit if you have a bond with the peer?